src/status_im/ui/screens/discover/events.cljs

```diff
@@ -1,148 +1,160 @@
-        (protocol/send-discoveries-response!
-          {:web3        web3
-           :discoveries (discoveries/get-all :asc)
-           :message     {:from current-public-key
-                         :to   to}})))))
-
-(register-handler :discoveries-request-received
-  (u/side-effect!
-    (fn [_ [_ {:keys [from]}]]
-      (dispatch [:discoveries-send-portions from]))))
-
-(register-handler :discoveries-response-received
-  (u/side-effect!
-    (fn [{:keys [discoveries]
-          :contacts/keys [contacts]} [_ {:keys [payload from]}]]
-      (when (get contacts from)
-        (when-let [data (:data payload)]
-          (doseq [{:keys [message-id] :as discover} data]
-            (when (and (not (discoveries/exists? message-id))
-                       (not (get discoveries message-id)))
-              (let [discover (assoc discover :created-at (time/now-ms))]
-                (dispatch [:add-discover discover])))))))))
-
-(defn add-discover
-  [db [_ discover]]
-  (assoc db :new-discover discover))
-
-(defn save-discover!
-  [{:keys [new-discover]} _]
-  (discoveries/save new-discover))
-
-(defn reload-tags!
-  [db _]
-  (assoc db :tags (discoveries/get-all-tags)
-            :discoveries (->> (discoveries/get-all :desc)
-                              (map (fn [{:keys [message-id] :as discover}]
-                                     [message-id discover]))
-                              (into {}))))
-
-(register-handler :add-discover
-  (u/handlers->
-    add-discover
-    save-discover!
-    reload-tags!))
-
-(register-handler :remove-old-discoveries!
-  (u/side-effect!
-    (fn [_ _]
-      (discoveries/delete :created-at :asc 1000 200))))
+(handlers/register-handler-fx
+  :broadcast-status
+  [(re-frame/inject-cofx :random-id)]
+  (fn [{{:keys [current-public-key web3]
+         :accounts/keys [accounts current-account-id]
+         :contacts/keys [contacts]} :db
+        random-id :random-id}
+       [_ status]]
+    (when-let [hashtags (seq (handlers/get-hashtags status))]
+      (let [{:keys [name photo-path]} (get accounts current-account-id)
+            message    {:message-id random-id
+                        :from       current-public-key
+                        :payload    {:message-id random-id
+                                     :status     status
+                                     :hashtags   (vec hashtags)
+                                     :profile    {:name          name
+                                                  :profile-image photo-path}}}]
+
+        {::broadcast-status {:web3       web3
+                             :message    message
+                             :identities (handlers/identities contacts)}
+         :dispatch          [:status-received message]}))))
+
+(handlers/register-handler-fx
+  :init-discoveries
+  (fn [_ _]
+    {:data-store.discover/get-all [#(re-frame/dispatch [:data-store/db-update {:discoveries %}])
+                                   maximum-number-of-discoveries]
```

Comment:
I don't get this very much. Why instead of getting all the discoveries as a simple co-effect which is then merged into app-db, we do this complicated thing ?
Once `:data-store.discover/get-all` effect could very well run before the `:request-discoveries` dispatch and once the effect is evaluated (ran), it will block till everything is red from the realm.

Reply:
In that particular case it could work but I think it is better to use the same mechanism for all queries. I don't understand the second part of your comment could you clarify ?

Reply:
You can ignore the second part, I thought the reason was performance, but it's not.
I just don't understand why instead of one co-effect, we have effect with callback + asynchronously dispatched handler, what's the benefit of such approach ?

Reply:
Because coeffects only work when you don't need parameters for your query, so I think it is better to use this approach consistently for all db queries rather than sometime use coeffects when it is possible

Reply:
Ok, I see the point now. 

Still I don't think it's quite worth it because, with this approach, it forces you to have extra events, one is `:init-discoveries` (or any other initialisation event), but you can't actually do anything related to the initial data in this event (because you don't have the loaded data at your disposal yet), the second is the `:data-store/db-update` event. I know it's generic, but there could be cases where it's not generic at all (you have to do some pre-processing before merging data into db).

I don't know if it's much better then using co-effects, even if sometimes instead of data, you have query function as a co-effect instead (when it needs to be parametrised).

Reply:
But if the coeffect returns a function that has a side-effect, how can the event handler be free of side-effects when you actually eval the function within ?

Reply:
The event handler in the strict sense is never free of side-effects (even in the ideal case where you can assign some data red from the external source into co-effects map).
The event handler as it is is composed of various interceptors and some of those necessary have to be impure, what is important is that the handler (last interceptor which you provide) is side-effect free (which it still is in a same sense as higher-order function like `map` or `reduce` are pure functions, even if you obviously can provide mapping/reducing fn with side effects).
The reasons why I think side-effect free handler functions are important are, in my opinion:

* That you can test them easily without having access to side effects available in real-env
* You can execute them in the repl, while app is running and not cause any side-effects on application (be it reading or writing)

Bot can be done by just providing deterministic function (like `(constantly something)` instead of the actual side-effect function registered in the normal env.

The other option, which is done in this PR basically says that we have to get rid of all coeffects for reading and always do the reading from persistent storage through effects and it always has to be asynchronous. 
It's the last point I'm quite uneasy with.